pyat/at/physics/linear.py

```diff
@@ -305,17 +217,25 @@ def analyze(r44):
         g = 1.0
 
     # Get initial twiss parameters
-    a0_a, b0_a, tune_a = _closure(A)
-    a0_b, b0_b, tune_b = _closure(B)
-    tune = numpy.array([tune_a, tune_b])
-
-    if get_chrom:
+    if twiin is None:
+        a0_a, b0_a, tune_a = _closure(A)
+        a0_b, b0_b, tune_b = _closure(B)
+        tune = numpy.array([tune_a, tune_b])
+
+    if get_chrom:      
+        if twiin is not None:
+            orbit[0:5] += numpy.append(disp0*0.5*ddp, 0.5 * ddp)
         d0_up, tune_up, _, l_up = linopt(ring, dp + 0.5 * ddp, uintrefs,
                                          keep_lattice=True,
-                                         coupled=coupled)
+                                         coupled=coupled, twiin=twiin)
```

Comment:
You are changing the initial orbit `orbit`, but it is not used. Probably it should be introduced in `twiin`?